### ISSUE TITLE
OPPO Find N2 Flip Compatibility

### ISSUE THREAD

**Issue by AbandonedCart** [SamSprung/SamSprung-TooUI#230]
It appears this is a portrait cover screen device that released in China on December 23, 2022 with a global release on February 15, 2023. Having heard about it just two days before the global launch, only minimal preparations were made. It appears a few YouTube channels cited success, but no issues were reported.

The first official changes to add support were published on February 21, 2023 (submitted on February 13, 2023) after being held up by a Google Play witch hunt for policy violations related to Accessibility. During this time, no issues or feedback have been received. Since no physical device is available for testing, the only information appears to be that the app works.

This issue is being opened for further analysis of compatibility.

**Comment by ystyle0616** [comment]
SamSprung TooIME keyboard not showing up

all set allow already

**Comment by ystyle0616** [comment]
Cover top screen have black bar space

**Comment by AbandonedCart** [comment]
Is the "black bar space" something that is part of the launcher or the launcher is not covering something on the screen? I don't quite understand what that means.

**Comment by ystyle0616** [comment]
[image: image.png]
THIS PART, not full screen

TwistedUmbrellaX ***@***.***> 于2023年3月6日周一 19:57写道：

**Comment by ystyle0616** [comment]
THIS PART, not full screen<a href="https://ibb.co/wg0Tbkf"><img src="https://i.ibb.co/H4F0MSj/cover.png" alt="cover" border="0"></a>

**Comment by AbandonedCart** [comment]
Looks like the last update I submitted is still in review. The latest changes set the proper resolution for the N2 cover. The app was originally designed for a screen that is 512 x 260, while the N2 cover is 720 x 382. If the changes already made don't resolve the issue, there is one other potential cause. It looks like the app is trying to leave room for an action bar that doesn't actually exist. I've changed the request for the screen size from the current size to the absolute maximum.

**Comment by AbandonedCart** [comment]
If you prefer not to wait for the Google Play review, the versions published in the releases on here are interchangeable. The update notices will switch to GitHub, but installing a Google Play version over it will change them back to Google Play.

https://github.com/SamSprung/SamSprung-TooUI/releases/tag/sideload

**Comment by ystyle0616** [comment]
OK  TQ , how about the keyboard , able to fix?

**Comment by ystyle0616** [comment]
After download new apk,  N2 cover still not full screen

**Comment by AbandonedCart** [comment]
Is it every app or just one particular one?

**Comment by ystyle0616** [comment]
Only system app work full screen.

Other than that,like google map, chatting app have black space on top screen

**Comment by AbandonedCart** [comment]
System apps working and third-party apps having a bar sounds like it is something the system is enforcing.

**Comment by ystyle0616** [comment]
WeChat,Instagram, ok, if app or game without topbar section will appear black,or able to toggle hide or unhide top section?

**Comment by AbandonedCart** [comment]
I'm not sure. Since apps aren't actually intended to run at all, it's not something that has an established solution.

**Comment by AbandonedCart** [comment]
From everything I've seen without having the phone myself, the issue seems to be that apps are treating the cover screen the same way they would treat the internal screen. It is running them with a section meant to respect the camera cutout and act as an actionbar buffer zone.

What this means is that it is a combination of system settings and possibly recycled code. The problem for using an app that bypasses the limitation on which apps can be used on the cover is that it still cannot override the system and app settings of the individual apps it runs. At best, it can "recommend" settings that are either used or ignored. 

On Samsung, going into the system's Settings app, Display, and Camera cutout allows modifying how the phone handles the front camera. I'd be interested to know if there is an equivalent setting on the N2 and if setting "Show camera cutout" for the apps with the black bar fixes them.

![cameracutout](https://user-images.githubusercontent.com/1173913/223419917-f7c299e8-76fe-460e-9589-46fff7d28880.png)

**Comment by AbandonedCart** [comment]
Any luck with that? If it worked, I will add a section to the guides about it

**Comment by ystyle0616** [comment]
Yes, dude its working! Choose show front camera to each app. still waiting fix for keyboard. then all feature unlock

and how about custom font size? able to adjust?


<a href="https://ibb.co/Xkj1NZx"><img src="https://i.ibb.co/Nnt5Q62/Image-20230308092612.jpg" alt="Image-20230308092612" border="0"></a>
<a href="https://ibb.co/ChGF9ry"><img src="https://i.ibb.co/pJFpKmT/Image-20230308092619.jpg" alt="Image-20230308092619" border="0"></a>

**Comment by ystyle0616** [comment]
All working. Left keyboard function hehe
<a href="https://ibb.co/cDMKmcb"><img src="https://i.ibb.co/sykpBv5/Image-20230308092612.jpg" alt="Image-20230308092612" border="0"></a>
<a href="https://ibb.co/2sSTqM9"><img src="https://i.ibb.co/b2Fk1v0/Image-20230308093315.jpg" alt="Image-20230308093315" border="0"></a>
<a href="https://ibb.co/QXQNcK9"><img src="https://i.ibb.co/1bQqnmZ/Image-20230308093319.jpg" alt="Image-20230308093319" border="0"></a>

**Comment by AbandonedCart** [comment]
The font size can't be adjusted for the launcher alone right now. It uses the system setting. I am still working on some flexibility for the scaling, but that is still a work in progress.

The keyboard is being worked on, but I have a few things to finish for another project.

**Comment by ystyle0616** [comment]
alright.Hope get your keyboard update, good news soon

**Comment by jheyare** [comment]
Yes keyboard is not working on Oppo N2 Flip. Hopefully, gets sorted soon. Thanks for your hardwork.

**Comment by jheyare** [comment]
I am having issues with the back button. I cant activate it on accessibility settings as it is restricted for security reasons. How do I go about that? Many thanks!

**Comment by AbandonedCart** [comment]
You can't. Unfortunately, Google Play felt a back button was not "necessary" and revoked permission.

**Comment by lerorin** [comment]
sorry if im being rude kinda new here but I've tested with my n2 flip. you need to allow the restricted settings under app info and then turn on the back function in the app. like in this guide

https://youtu.be/0bcLjpfrmHw?t=70

so far i have no issue with back button aside from the bar is a bit to the right so its a bit hard to press sometimes due to my phone case.

**Comment by AbandonedCart** [comment]
I think my wording may have caused some confusion. I know **how** to enable back button functionality. It's been a feature for over a year now. On a whim, Google Play said I either remove it or they will remove the app because they consider the back button to be an unnecessary feature.

**Comment by Pedned85** [comment]
Hello All,

Apologies for my naivety here, i literally stumbled across this app on Reddit where someone had installed the UI to utilise the front cover screen on the N2 Flip - i found this in the Play store and downloaded, accepted all the permissions for it however i dont appear to have any functionality at all, my cover screen cant be operated at all, i just see my wallpaper.

Im obviously doing something wrong here so hoping someone can help me out, either comment or DM just so i can get this working. Ive uninstalled for now over concerns that i may be damaging my device.

Thanks in advance for any assistance

**Comment by AbandonedCart** [comment]
This issue is for trying to resolve compatibility issue. General instructions are at https://samsprung.github.io/launcher/index.html

**Comment by ystyle0616** [comment]
[Pedned85] remove app and install again. Accept each permission. Mainly switch on/off (top right button)
1. Let SamSprung Launcher Open (dont switch or kill app)
2. then close your cover.
3. unlock with your fingerprint
4. tap bottom right corner(small icon) to active app launcher
5. Choose app and open

**Comment by AbandonedCart** [comment]
The page I linked starts with install and setup, but also goes into detail with pictures.

**Comment by ystyle0616** [comment]
version 2.4.0 
This option missing (and make screen stuck)
 bottom drawer minimize button can be set to close TooUI until the screen is turned off

Version 2.3.9 everything work fine,include back button function

**Comment by AbandonedCart** [comment]
The setting was removed. As stated in the reply to your email, it was replaced with double tapping the launcher button.

**Comment by ystyle0616** [comment]
Now having this issue
[https://vimeo.com/808907893](url)

**Comment by AbandonedCart** [comment]
You have to double tap directly on the icon, not above it.

**Comment by ystyle0616** [comment]
problem is app list wont show up...

**Comment by AbandonedCart** [comment]
Tap it once for the list.

**Comment by ystyle0616** [comment]
yup....tap once its stuck, double tap few time drawer close

**Comment by AbandonedCart** [comment]
Added an option to use a larger icon. That should make it easier.

**Comment by ystyle0616** [comment]
thank buddy , problem fix. notice ignore battery no show up in the setting. 2.3.X version was ok

**Comment by AbandonedCart** [comment]
OPPO broke that one. Nothing has changed with it. Fortunately, you don't really need it except in some rare cases where some app is using a ton of memory

**Comment by ystyle0616** [comment]
hmmm without this, app will be auto kill

TwistedUmbrellaX ***@***.***> 于2023年3月17日周五 11:48写道：

**Comment by JordanSpringerNature** [comment]
Any updated/news on whether the keyboard (Samsprung-ToolME) will be working with the Oppo N2 flip?
I downloaded it and it was working for a couple days and now will no longer trigger with the input fields - uninstalled and re-installed everything and still can't get it to trigger again?

**Comment by AbandonedCart** [comment]
Not having an N2 Flip on site, there is no way to determine the cause of any issue based on general statements. It has to wait for someone to report the issue with a log that shows the cause.

**Comment by JordanSpringerNature** [comment]
Ah okay - I did just log an issue but failed to paste some logs - let me figure out how to do that.....

**Comment by cabban** [comment]
My banking app not working when activated samsprung it says dangerous malware cause is not in Google play :( its working awesome samsprung can u upload it to work with Oppo flips🫠and banking app

**Comment by AbandonedCart** [comment]
SamSprung is in Google Play, though.

**Comment by cabban** [comment]
> SamSprung is in Google Play, though.

Nah otherwise i would wrote its in Google play not available for my phone anyway not available for this version it says i cant download it , Cheers

**Comment by AbandonedCart** [comment]
> Nah otherwise i would wrote its in Google play not available for my phone anyway not available for this version it says i cant download it , Cheers

https://play.google.com/store/apps/details?id=com.eightbit.samsprung

**Comment by cabban** [comment]
> > Nah otherwise i would wrote its in Google play not available for my phone anyway not available for this version it says i cant download it , Cheers
> 
> https://play.google.com/store/apps/details?id=com.eightbit.samsprung

As i told you not available for my Oppo Android 14 
![Screenshot_2024-09-16-17-31-25-86_b5a5c5cb02ca09c784c5d88160e2ec24](https://github.com/user-attachments/assets/2328af6c-0fda-4558-bc8f-dd4665d97a50)

**Comment by AbandonedCart** [comment]
What locale are you? It shouldn’t be blocked for you, so I will have to contact Play about it.

**Comment by cabban** [comment]
> What locale are you? It shouldn’t be blocked for you, so I will have to contact Play about it.

Sweden thank u

**Comment by AbandonedCart** [comment]
No worries. Sorry for the confusion.

**Comment by cabban** [comment]
> No worries. Sorry for the confusion.

Its ok 👍 ;)

**Comment by AbandonedCart** [comment]
Should pop up within the next day or so. I’m not sure how long it takes to refresh on their end. Looks like a couple OPPO devices got excluded by mistake.

**Comment by cabban** [comment]
> Should pop up within the next day or so. I’m not sure how long it takes to refresh on their end. Looks like a couple OPPO devices got excluded by mistake.

Thank u :), can i request some stuff :) ?

**Comment by AbandonedCart** [comment]
You may want to start a new issue for it and I don’t have an OPPO, so a lot of it is trial and error.

**Comment by cabban** [comment]
> You may want to start a new issue for it and I don’t have an OPPO, so a lot of it is trial and error.

Now all is working that not working when sideloaded downloaded from Google play ;)

**Comment by cabban** [comment]
> You may want to start a new issue for it and I don’t have an OPPO, so a lot of it is trial and error.

I have a big issue the outer screen is freezing cant answer calls swipe or remove stock notification can you check :)

**Comment by AbandonedCart** [comment]
Did you double tap the launcher icon to disable it?

There are instructions for the app at https://samsprung.github.io/launcher/

**Comment by cabban** [comment]
> Did you double tap the launcher icon to disable it?
> 
> There are instructions for the app at https://samsprung.github.io/launcher/

Aha no i will try :) next time :)

**Comment by cabban** [comment]
Can we have the icons and clock showing now its half

**Comment by AbandonedCart** [comment]
I don't understand the question
